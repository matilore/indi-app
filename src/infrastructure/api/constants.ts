import { Podcast } from "@/domain/interfaces";

export const PODCAST_NUMBER = 100;

export const REACT_APP_API_PODCASTS_URL = `https://itunes.apple.com/us/rss/toppodcasts/limit=${PODCAST_NUMBER}/genre=1310/json`;

export const MOCKED_RESPONSE: Podcast[] = [
  {
    "im:name": {
      label: "The Joe Budden Podcast",
    },
    "im:image": [
      {
        label:
          "https://is3-ssl.mzstatic.com/image/thumb/Podcasts113/v4/f2/21/fa/f221fabd-017f-5125-633b-f1fe4f39802a/mza_182995249085044287.jpg/55x55bb.png",
        attributes: {
          height: "55",
        },
      },
      {
        label:
          "https://is5-ssl.mzstatic.com/image/thumb/Podcasts113/v4/f2/21/fa/f221fabd-017f-5125-633b-f1fe4f39802a/mza_182995249085044287.jpg/60x60bb.png",
        attributes: {
          height: "60",
        },
      },
      {
        label:
          "https://is1-ssl.mzstatic.com/image/thumb/Podcasts113/v4/f2/21/fa/f221fabd-017f-5125-633b-f1fe4f39802a/mza_182995249085044287.jpg/170x170bb.png",
        attributes: {
          height: "170",
        },
      },
    ],
    summary: {
      label:
        "Tune into Joe Budden and his friends. Follow along the crazy adventures of these very random friends.",
    },
    "im:price": {
      label: "Get",
      attributes: {
        amount: "0",
        currency: "USD",
      },
    },
    "im:contentType": {
      attributes: {
        term: "Podcast",
        label: "Podcast",
      },
    },
    rights: {
      label: "Â© All rights reserved",
    },
    title: {
      label: "The Joe Budden Podcast - The Joe Budden Network",
    },
    link: {
      attributes: {
        rel: "alternate",
        type: "text/html",
        href: "https://podcasts.apple.com/us/podcast/the-joe-budden-podcast/id1535809341?uo=2",
      },
    },
    id: {
      label:
        "https://podcasts.apple.com/us/podcast/the-joe-budden-podcast/id1535809341?uo=2",
      attributes: {
        "im:id": "1535809341",
      },
    },
    "im:artist": {
      label: "The Joe Budden Network",
      attributes: {
        href: "https://podcasts.apple.com/us/artist/the-joe-budden-network/1535844019?uo=2",
      },
    },
    category: {
      attributes: {
        "im:id": "1310",
        term: "Music",
        scheme:
          "https://podcasts.apple.com/us/genre/podcasts-music/id1310?uo=2",
        label: "Music",
      },
    },
    "im:releaseDate": {
      label: "2023-06-28T00:00:00-07:00",
      attributes: {
        label: "June 28, 2023",
      },
    },
  },
];
