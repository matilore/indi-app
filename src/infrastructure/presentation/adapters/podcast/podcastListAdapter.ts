import { Podcast } from "@/domain/interfaces";
import { PodcastList } from "./contracts";

export class PodcastListAdapter {
  private podcastFromApi: Podcast;

  constructor(podcastFromApi: Podcast) {
    this.podcastFromApi = podcastFromApi;
  }

  get img() {
    return this.podcastFromApi["im:image"][2].label;
  }

  get title() {
    return this.podcastFromApi["im:name"].label;
  }

  get author() {
    return this.podcastFromApi["im:artist"].label;
  }

  toJSON(): PodcastList {
    return {
      img: this.img,
      title: this.title,
      author: this.author,
    };
  }
}
