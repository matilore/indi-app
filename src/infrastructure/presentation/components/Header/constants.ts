export const LINK_TEXT = "Podcaster";
