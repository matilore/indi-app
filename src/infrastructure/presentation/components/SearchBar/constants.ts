export const SEARCH_BAR_PLACEHOLDER = "Filter podcasts...";

export const PODCAST_NUMBER_BADGE_ID = "podcast-number-badge";
