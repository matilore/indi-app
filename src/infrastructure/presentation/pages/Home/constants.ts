export const LOCAL_STORAGE_ITEM_NAME = "podcasts";
