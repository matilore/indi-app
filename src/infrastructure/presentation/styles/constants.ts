const BLACK = "#000000";
const GREY = "#6A6F85";
const WHITE = "#FFFFFF";
const LIGHT_GREY = "#FAFAFC;";

export const COLORS = {
  BLACK,
  GREY,
  WHITE,
  LIGHT_GREY,
};

export const FONT_SIZES = {
  EXTRA_SMALL: "12px",
  SMALL: "14px",
  MEDIUM: "16px",
  LARGE: "20px",
  EXTRA_LARGE: "24px",
};

export const FONT_SIZES_TYPES = {
  EXTRA_SMALL: "extraSmall",
  SMALL: "small",
  MEDIUM: "medium",
  LARGE: "large",
  EXTRA_LARGE: "extraLarge",
};
