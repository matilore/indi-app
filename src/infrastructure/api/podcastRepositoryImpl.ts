import { REACT_APP_API_PODCASTS_URL } from "./constants";
import {
  PodcastRepository,
  Podcast,
  PodcastsApiResponse,
} from "@/domain/interfaces";

export class PodcastRepositoryImpl implements PodcastRepository {
  async getAllPodcasts(): Promise<Podcast[]> {
    const res = await fetch(REACT_APP_API_PODCASTS_URL);
    const response: PodcastsApiResponse = await res.json();
    return response.feed.entry;
  }
}
