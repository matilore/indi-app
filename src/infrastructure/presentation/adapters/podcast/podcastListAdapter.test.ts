import { PodcastListAdapter } from "./podcastListAdapter";
import { MOCKED_RESPONSE } from "@/infrastructure/api/constants";
describe("PodcastListAdapter", () => {
  it("should convert to JSON format correctly", () => {
    const podcastUi = new PodcastListAdapter(MOCKED_RESPONSE[0]);
    const json = podcastUi.toJSON();

    expect(json).toMatchSnapshot();
  });
});
