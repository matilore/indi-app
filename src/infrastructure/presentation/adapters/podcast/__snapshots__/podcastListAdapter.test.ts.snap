// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PodcastListAdapter > should convert to JSON format correctly 1`] = `
{
  "author": "The Joe Budden Network",
  "img": "https://is1-ssl.mzstatic.com/image/thumb/Podcasts113/v4/f2/21/fa/f221fabd-017f-5125-633b-f1fe4f39802a/mza_182995249085044287.jpg/170x170bb.png",
  "title": "The Joe Budden Podcast",
}
`;
